src/run.rs

```diff
@@ -0,0 +1,171 @@
+//! CLI tool to compile Cranelift IR files to native code in memory and execute them.
+
+use crate::utils::read_to_string;
+use core::mem;
+use cranelift_codegen::binemit::Reloc;
+use cranelift_codegen::ir::{ExternalName, Function, JumpTable};
+use cranelift_codegen::isa::TargetIsa;
+use cranelift_codegen::print_errors::pretty_error;
+use cranelift_codegen::Context;
+use cranelift_codegen::{binemit, ir, settings};
+use cranelift_native::builder as host_isa_builder;
+use cranelift_reader::{parse_test, Details};
+use region::{protect, Protection};
+use std::path::PathBuf;
+
+pub fn run(files: Vec<String>, flag_print: bool) -> Result<(), String> {
+    let isa = create_target_isa();
+    let mut total = 0;
+    let mut errors = 0;
+    for file in iterate_files(files) {
+        total += 1;
+        let file_name = get_file_name(&file)?;
```

Comment:
Please use the `path` directly instead to prevent ambiguity when multiple dirs contain the same filename.